src/app/content/components/utils/attachHighlight.ts

```diff
@@ -0,0 +1,18 @@
+import Highlighter, { Highlight, SerializedHighlight } from '@openstax/highlighter';
+import Sentry from '../../../../helpers/Sentry';
+
+const attachHighlight = (
+  highlight: Highlight | SerializedHighlight,
+  highlighter: Highlighter
+) => {
+  if (highlight instanceof SerializedHighlight) {
```

Comment:
you don't have to recurse here, it does it internally

Reply:
@TomWoodward It does, but `SerializedHighlight` has no `isAttached` method. I could write this as: (is this better)?
```
  highlighter.highlight(highlight);
  const hl = highlight instanceof SerializedHighlight
    ? highlighter.getHighlight(highlight.id)
    : highlight;
  if (!hl || !hl.isAttached()) {
    Sentry.captureException(`Highlight with id: ${highlight.id} has not been attached.`);
  }
```

Reply:
I've updated the code, but it looks like some tests from `highlightManager.spec.tsx` are failing
Probably some mocks needs updates

Reply:
src/app/content/components/Page/highlightManager.ts already calls highlighter.getHighlight(data.id); after it calls highlight, you could have your attachHighlight utility do that and return the highlight to save it the trouble